src/App.tsx

```diff
@@ -1,18 +1,42 @@
-import React from 'react';
+import React, { ChangeEvent, useState } from 'react';
 import './App.css';
+import { Pagination } from './components/Pagination';
 import { getNumbers } from './utils';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42)
   .map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [currentPage, setCurrentPage] = useState(1);
+  const [itemsPerPage, setItemsPerPage] = useState(5);
+
+  const startPage = (currentPage - 1) * itemsPerPage;
+  let endPage = currentPage * itemsPerPage;
+
+  if (endPage > items.length) {
+    endPage = items.length;
+  }
```

Comment:
it would be better to simplify
```suggestion
  const endPage = endPage > items.length ? items.length : currentPage * itemsPerPage;
```